woocommerce/woocommerce-gutenberg-products-block
# Add E2E tests for Filter blocks (Price, Attribute, and Active Filters)

Closes #4643


This PR will introduce E2E tests for the `Filter Products by Price` block.

The changes included are:
- Add a class to the `div` containing the Filter Products by Price block in the editor (`wp-block-woocommerce-price-filter`). Previously there was not one so our testing utils that rely on class names couldn't target the block.
- Add a class to the `div` containing the Filter Products by Attribute block in the editor (`wc-block-attribute-filter`).
- Add [`pptr-testing-library`](https://www.npmjs.com/package/pptr-testing-library) so we can use the queries provided by it.
- Add `clearAndFillInput` util so we can type into input boxes.
- Add attributes (Capacity and Shade) to the E2E test store.
- Create two new products that use the Capacity attribute.
- Ensure attributes are added correctly to the product by ID (see https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/4562/files#diff-2fccf9f205d4c46e7393da8019992bb5bd730aa2a3e854420bba4cdf661ee3ffR362-R398) - This creates the attributes in the store then returns their IDs. We can't know ahead of time what the ID will be so it needs to be fetched from the API after creation.
- Add tests for `Active Filters`, `Filter Products by Price`, and `Filter Products by Attribute`.


See #2395 huge thanks to @senadir for the initial work done in that PR 🙌🏼 

### How to test the changes in this Pull Request:

1. Ensure the Filter Products by Price block works in the editor and on the front-end still.
2. Ensure the Filter Products by Attribute block works in the editor and on the front-end still.
2. Run the E2E tests - verify all pass.
3. Wait for tests to go green here ✅ 



### Changelog

> Add end-to-end tests to cover the `Active Filters`, `Filter Products by Attribute`, and `Filter Products by Price` blocks.
